### PR TITLE
Document a proposal for the storage UI

### PR DESCRIPTION
**Pull request by ancorgs**
We need a way to define in the D-Installer UI which partitions and LVM logical volumes should be created and how to make space for them if we are not installing in an empty disk.

We need something that is:

- As automatic as the YaST Guided Setup
- Easier to understand than YaST Guided Setup
- Almost as flexible as the YaST Expert Partitioner

This pull request documents a proposal for such an UI. For better review, use the Github option "View File" on `doc/storage_ui.md`.

### CODE REVIEW

File: doc/storage_ui.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+aspects like the file-system type or the usage of snapshots and subvolumes.
+
+Last but not least, the table header includes a small description about the origin of the current
+list of volumes. Maybe some additional details can be displayed with a tool-tip or maybe that small
+description can be displayed by any other mechanism. Next to the description a set of two grouped
+actions allow to:
+
+- Reset the list to the default volumes using their recommended sizes (i.e. desired) as lower limit
+- Same action but using the minimum sizes (i.e. min) as lower limit
+
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply doc/storage_ui.md:153]: I haven't though yet how to cover that use-case. But it makes way more sense to me to present the option in the "target" part (or in a completely different place) than as an option in the table of volumes. In my opinion, it doesn't fit there.

File: doc/storage_ui.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+The size of each volume is specified as a couple of lower and upper limits (the upper one is
+optional in all cases). With the current approach of the YaST `GuidedProposal` there some cases in
+which the lower and upper limits can be adjusted based on the following aspects:
+
+- Whether snapshots are activated for the root volume
+- Whether the size of the volume must be influenced by the RAM size (used for suspend in the case
+  of swap and for Kdump in the case of the root volume)
+- Whether the given volume is marked as "fallback" for another one (eg. if the separate /home is
+  disabled then the upper limit of the root one dissapears)
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply doc/storage_ui.md]: I added this thread as a section to the document.

File: doc/storage_ui.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ancorgs) [reply doc/storage_ui.md:142]: Added a not about that use-case to the document.

File: doc/storage_ui.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+the future by a better alternative to show the result.
+
+### Calculating How to Make Space
+
+Although D-Installer reuses part of the internal logic of YaST, there is no need to reproduce all
+YaST behaviors.
+
+YaST's `GuidedProposal` contains a subcomponent called `SpaceMaker` which takes cares of deleting
+and resizing existing partitions to make space for the new system. It decides by itself which
+partitions to should be affected following a logic that, even though is configurable in the control
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply doc/storage_ui.md]: Fixed.

**Review by dgdavid — APPROVED** [review]
> ### Don't Take Mock-ups Too Seriously
> First of all, bear in mind the screenshots are far from being a faithfull representation of the
final look & feel. This document presents the concept focusing on the elements that should be there and how they will interact.

So, LGTM :stuck_out_tongue_winking_eye:

**Pull request merged**